src/Runner.Worker/Handlers/CompositeActionHandler.cs

```diff
@@ -295,98 +295,109 @@ private async Task RunStepsAsync(List<IStep> embeddedSteps, ActionRunStage stage
                 CancellationTokenRegistration? jobCancelRegister = null;
                 try
                 {
-                    // Register job cancellation call back only if job cancellation token not been fire before each step run
-                    if (!ExecutionContext.Root.CancellationToken.IsCancellationRequested)
+                    // For main steps just run the action
+                    if (stage == ActionRunStage.Main)
```

Comment:
This just restores the old behavior for main steps found here: https://github.com/actions/runner/blob/419ed24c1ee8c70eeb096570980e700e833f81f6/src/Runner.Worker/Handlers/CompositeActionHandler.cs#L194

Reply:
discussed offline. When we support condition on Main in the future, we will need to register the cancel-callback. For now we don't need it.